tests/routers/v1/application_test.py

```diff
@@ -132,14 +126,31 @@ def test_post_application_returns_duplicate_error_if_id_already_exists(
     assert response.status_code == 409
 
 
-@patch("laa_crime_application_store_app.internal.notifier.Notifier.notify")
+def test_post_application_notifies_subscribers(
+    client: TestClient, dbsession: Session, seed_subscriber, httpx_mock
+):
+    httpx_mock.add_response(url=seed_subscriber.webhook_url)
```

Comment:
I am likely being dim but not fully sure I understand how this works.
https://pypi.org/project/pytest-httpx/ says that any response that does not come back will mean the test fails but the examples seem to suggest that you are looking at the response from whatever you do next, ie the client.post. 

Or is it that you get two responses back from this post (our one and the one to the webhook) and you have limited it to checking the webhook one?

> In case more than one response match request, the first one not yet sent (according to the registration order) will be sent.
> 
> In case all matching responses have been sent, the last one (according to the registration order) will be sent.
> 
> You can add criteria so that response will be sent only in case of a more specific matching.
>

Reply:
If that is the case then a comment to explain how the test fails would be good.

Reply:
So basically there's a hidden assertion as far as I can tell. This line says "If by the end of the test there has been no call made to `seed_subscriber.webhook_url`, throw a wobbly and cause the test to fail." You're right, a comment making this clear makes sense - it wasn't at all clear to me until I deliberately failed the test, but then again _nothing_ in python is clear to me!